idp-service/src/test/java/io/mosip/idp/services/AuthorizationServiceTest.java

```diff
@@ -409,4 +378,182 @@ public void getOauthDetails_withValidClaimsInDbAndValidClaimsInReqAndNoOPENIDSco
             Assert.assertTrue(ex.getErrorCode().equals(ErrorConstants.INVALID_SCOPE));
         }
     }
+
+    @Test
+    public void authenticate_withInvalidTransaction_thenFail() {
+        String transactionId = "test-transaction";
+        when(cacheUtilService.getPreAuthTransaction(transactionId)).thenReturn(null);
+
+        KycAuthRequest kycAuthRequest = new KycAuthRequest();
+        kycAuthRequest.setTransactionId(transactionId);
+        try {
+            authorizationServiceImpl.authenticateUser(kycAuthRequest);
+            Assert.fail();
+        } catch (IdPException ex) {
+            Assert.assertTrue(ex.getErrorCode().equals(ErrorConstants.INVALID_TRANSACTION));
+        }
+    }
+
+    @Test
+    public void authenticate_multipleRegisteredAcrsWithSingleFactor_thenPass() throws IdPException, KycAuthException {
+        String transactionId = "test-transaction";
+        when(cacheUtilService.getPreAuthTransaction(transactionId)).thenReturn(createIdpTransaction(
+                new String[]{"mosip:idp:acr:generated-code", "mosip:idp:acr:static-code"}));
+
+        List<List<AuthenticationFactor>> allAuthFactors=new ArrayList<>();
+        allAuthFactors.add(getAuthFactors("mosip:idp:acr:generated-code"));
+        allAuthFactors.add(getAuthFactors("mosip:idp:acr:static-code"));
+        when(authenticationContextClassRefUtil.getAuthFactors(new String[]{"mosip:idp:acr:generated-code",
+                "mosip:idp:acr:static-code"})).thenReturn(allAuthFactors);
+
+        KycAuthResult kycAuthResult = new KycAuthResult();
+        kycAuthResult.setKycToken("test-kyc-token");
+        kycAuthResult.setPartnerSpecificUserToken("test-psut");
+        when(authenticationWrapper.doKycAuth(anyString(), anyString(), any())).thenReturn(kycAuthResult);
+
+        KycAuthRequest kycAuthRequest = new KycAuthRequest();
+        kycAuthRequest.setTransactionId(transactionId);
+        kycAuthRequest.setIndividualId("23423434234");
+        List<AuthChallenge> authChallenges = new ArrayList<>();
+        authChallenges.add(getAuthChallengeDto("OTP"));
+        kycAuthRequest.setChallengeList(authChallenges);
+
+        AuthResponse authResponse = authorizationServiceImpl.authenticateUser(kycAuthRequest);
+        Assert.assertNotNull(authResponse);
+        Assert.assertEquals(transactionId, authResponse.getTransactionId());
+    }
+
+    @Test
+    public void authenticate_multipleRegisteredAcrsWithInvalidSingleFactor_thenPass() throws IdPException {
```

Comment:
this looks like a negative case, why is this called pass ?

Reply:
Corrected.